share/config.fish

```diff
@@ -152,3 +152,15 @@ function . --description 'Evaluate contents of file (deprecated, see "source")'
 		source $argv
 	end
 end
+
+# As last part of initialization, source the snippets directories
+# Implement precedence (User > Admin > Vendors > Fish) by basically doing "basename"
+set -l sourcelist
+for file in $configdir/fish/snippets/* $__fish_sysconfdir/snippets/* $__fish_datadir/vendor_snippets.d/*
```

Comment:
This is pretty clever, but we'll have to watch the case where those wildcards expand to nothing once #2394 is changed.